### ISSUE TITLE
Unable to specify a configuration File

### ISSUE THREAD

**Issue by trevormerritt** [hexparrot/mineos-node#413]
Currently, paths are hardcoded.
I don't want the flexibility to run the software from a different location.

* I'm currently working on this.

**Comment by hexparrot** [comment]
What locations are you hoping to run it from that aren't currently working? 

I believe I've successfully had it in user `/home` trees, `/root` `/opt`, `/usr/games/minecraft`, `/usr/local/games/minecraft`, and I don't see any reason why it couldn't be anywhere, provided that all the sub-directory structure is intact. Let me know what you're thinking, or I also can just wait for the PR.

Look forward to it!

**Comment by trevormerritt** [comment]
https://github.com/hexparrot/mineos-node/blob/master/webui.js#L150

This line is where the configuration file being used is set.
At this point, there are 2 options for configuration file:

 - /etc/mineos.conf
 - /usr/local/etc/mineos.conf
  
 Given that the ports used by the services spun up are above 1024, there is really no need for root, nor for writing to these directories.  
 
 If the app is going to run in /home/randomuser/minecraft/myserver10122/thisis_The_coolest
 The user *should* be able to specify the config file as say
  /home/randomuser/minecraft/myserver10112/coolest.conf
  
  I looked at submitting a PR, but I've not used the async lib in years and to bring it to promises, would be more then I'm interested in.  I wish you good luck.
  
  The No-Call-Home PR is still waiting.  :)

**Comment by flareofghast** [comment]
Not sure what you're trying to achieve. The config defines the webserver settings and the ssl and other things. The location is fairly standard. Are you trying to have multiple users having their own webserver, ssl and servers?

**Comment by hexparrot** [comment]
I don't think that you'd need any big changes to the code to support changing the path of the configuration file.


`var mineos_config = read_ini('/etc/mineos.conf') || read_ini('/usr/local/etc/mineos.conf') || {};`
TO
`var mineos_config = read_ini('custom.conf') || read_ini('/etc/mineos.conf') || read_ini('/usr/local/etc/mineos.conf') || {};`

Or something to that effect.  The reason the filepath has to be hardcoded, is I need to create _some_ convention, and following `/etc` was the most straightforward.  Naturally, `./mineos.conf` would be problematic because of it being replaced on git pulls.

Allowing users to be able to run this in fullness from their respective, unprivileged areas, I can completely understand that. Let me see what I can write up.  I think it's just gonna be ^.

**Comment by hexparrot** [comment]
I've made https://github.com/hexparrot/mineos-node/pull/425

It works, but with a pretty hacky solution of giving unpriv'ed users access to `/etc/shadow`.  Need to figure out how to do auth without `sudo`, etc.
